sql/tests: TestRandomSyntaxGeneration failed

sql/tests.TestRandomSyntaxGeneration [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21299097?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_RandomSyntaxTestsBazel/21299097?buildTab=artifacts#/) on release-24.1 @ [3e45d29085b51fb027ab56d2766d0c8611452f27](https://github.com/cockroachdb/cockroach/commits/3e45d29085b51fb027ab56d2766d0c8611452f27):


```
*   | runtime.sigpanic
*   | 	GOROOT/src/runtime/signal_unix.go:881
*   | github.com/cockroachdb/cockroach/pkg/sql/gcjob.refreshTenant
*   | 	github.com/cockroachdb/cockroach/pkg/sql/gcjob/refresh_statuses.go:465
*   | github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.legacyWaitAndClearTableData
*   | 	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:530
*   | github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.Resume
*   | 	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:374
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1593
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1594
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:451
*   | github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1
*   | 	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:285
*   | github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2
*   | 	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:480
*   | runtime.goexit
*   | 	src/runtime/asm_amd64.s:1695
* Wraps: (2) runtime error: invalid memory address or nil pointer dereference
* Error types: (1) *withstack.withStack (2) runtime.errorString
*
panic: runtime error: invalid memory address or nil pointer dereference [recovered]
	panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x10 pc=0x4fb0cbf]

goroutine 2114943 [running]:
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).recover(0x119a?, {0x824d150, 0xc007a516e0})
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:226 +0x65
panic({0x5f207a0?, 0xbda8170?})
	GOROOT/src/runtime/panic.go:770 +0x132
github.com/cockroachdb/cockroach/pkg/sql/gcjob.refreshTenant({0x824d150, 0xc007a51a70}, 0xc00792e008, 0x18a3acc248b5b17c, 0xc5f0720?, 0xc00cab3480)
	github.com/cockroachdb/cockroach/pkg/sql/gcjob/refresh_statuses.go:465 +0x13f
github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.legacyWaitAndClearTableData({_}, {_, _}, {0xc002214000, {0xc0046d25b0, 0xc001818040, 0xc0046d25a0, 0xc00464bae0}, {{0xc00169c6f0, 0xc00169c708}, ...}, ...}, ...)
	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:530 +0x3c7
github.com/cockroachdb/cockroach/pkg/sql/gcjob.schemaChangeGCResumer.Resume({0xc021a309c0}, {0x824d150, 0xc007a51a70}, {0x65dfea0?, 0xc00b9cd180?})
	github.com/cockroachdb/cockroach/pkg/sql/gcjob/gc_job.go:374 +0x2b6
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine.func2(0xc021a309c0?, 0x0?, 0xc01491f918, {0x8245940?, 0xc00b9cd158?}, {0x824d150?, 0xc007a51a70?}, {0x65dfea0?, 0xc00b9cd180?})
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1593 +0xbb
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).stepThroughStateMachine(0xc0013d8588, {0x824d150, 0xc007a519b0}, {0x65dfea0, 0xc00b9cd180}, {0x8245940, 0xc00b9cd158}, 0xc021a309c0, {0xc02839bcb0, 0x7}, ...)
	github.com/cockroachdb/cockroach/pkg/jobs/registry.go:1594 +0xaa5
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).runJob(0xc0013d8588, {0x824d150, 0xc007a519b0}, {0x8245940, 0xc00b9cd158}, 0xc021a309c0, {0xc02839bcb0, 0x7}, {0xc01805c528, 0x17})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:451 +0x55a
github.com/cockroachdb/cockroach/pkg/jobs.(*Registry).resumeJob.func1({0x824d150?, 0xc007a516e0?})
	github.com/cockroachdb/cockroach/pkg/jobs/adopt.go:285 +0x13c
github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx.func2()
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:480 +0x13a
created by github.com/cockroachdb/cockroach/pkg/util/stop.(*Stopper).RunAsyncTaskEx in goroutine 2114902
	github.com/cockroachdb/cockroach/pkg/util/stop/stopper.go:471 +0x3fe
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<details><summary>Same failure on other branches</summary>
<p>

- #167545 sql/tests: TestRandomSyntaxGeneration failed [C-test-failure O-robot T-sql-foundations branch-master release-blocker]
- #132558 sql/tests: TestRandomSyntaxGeneration failed [CREATE TENANT statement is not valid after parsing then formatting] [C-test-failure O-robot P-3 T-db-server branch-release-24.1.6-rc]
</p>
</details>
/cc @cockroachdb/sql-foundations
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestRandomSyntaxGeneration.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/bazci/githubpost/issues)

</sub>


Jira issue: CRDB-62535

--- a/pkg/sql/gcjob/refresh_statuses.go
+++ b/pkg/sql/gcjob/refresh_statuses.go
@@ -284,7 +284,7 @@ func getIndexTTL(tableTTL int32, placeholder *zonepb.ZoneConfig, indexID descpb.
 
 func getTableTTL(defTTL int32, zoneCfg *zonepb.ZoneConfig) int32 {
 	ttlSeconds := defTTL
-	if zoneCfg != nil {
+	if zoneCfg != nil && zoneCfg.GC != nil {
 		ttlSeconds = zoneCfg.GC.TTLSeconds
 	}
 	return ttlSeconds
@@ -426,7 +426,7 @@ func refreshTenant(
 	tenantTTLSeconds := execCfg.DefaultZoneConfig.GC.TTLSeconds
 	zoneCfg, err := cfg.GetZoneConfigForObject(keys.SystemSQLCodec, keys.TenantsRangesID)
 	if err == nil {
-		tenantTTLSeconds = zoneCfg.GC.TTLSeconds
+		tenantTTLSeconds = getTableTTL(tenantTTLSeconds, zoneCfg)
 	} else {
 		log.Dev.Errorf(ctx, "zone config for tenants range: err = %+v", err)
 	}

